std/typecons.d

```diff
@@ -4628,6 +4640,63 @@ unittest
     //func(n);
 }
 
+///The proxied value must be an $(B lvalue).
+unittest
+{
+    struct NewIntType
+    {
+        //Won't work; the literal '1' is
+        //is an rvalue, not an lvalue
+        //mixin Proxy!1; 
+        
+        //Okay, n is an lvalue
+        int n;
+        mixin Proxy!n;
+        
+        this(int n) { this.n = n; }
+    }
+    
+    NewIntType nit = 0;
+    nit++;
+    assert(nit == 1);
+    
+    
+    struct NewObjectType
+    {
+        //Ok, "new Object" is an lvalue.
+        //This code is a bit weird, though.
+        mixin Proxy!(new Object);
+    }
+    
+    NewObjectType not;
+    assert(__traits(compiles, not.toHash()));
+}
+
+/**
+    There is one exception to the fact that the new type is not related to the
+    old type. Pseudo member functions that use 
+    $(LINK2 http://ddili.org/ders/d.en/ufcs.html, Universal Function Call Syntax)
```

Comment:
There's no reason to link to third-party resources when the official ones suffice.

http://dlang.org/function.html#pseudo-member